services/src/main/java/org/keycloak/validation/DefaultClientValidationProvider.java

```diff
@@ -264,4 +269,20 @@ private void validateJwks(ValidationContext<ClientModel> context) {
             context.addError("jwksUrl", "Illegal to use both jwks_uri and jwks_string", "duplicatedJwksSettings");
         }
     }
+
+    private void validateDefaultAcrValues(ValidationContext<ClientModel> context) {
+        ClientModel client = context.getObjectToValidate();
+        List<String> defaultAcrValues = AcrUtils.getDefaultAcrValues(client);
+        for (String configuredAcr : defaultAcrValues) {
+            Map<String, Integer> acrToLoaMap = AcrUtils.getAcrLoaMap(client);
+            if (acrToLoaMap.isEmpty()) {
+                acrToLoaMap = AcrUtils.getAcrLoaMap(client.getRealm());
```

Comment:
There is an another option that it ignores client's acr to loa mapping the same as done in https://github.com/keycloak/keycloak/pull/10265 .
Server metadata `acr_values_supported ` does not contain client's acr to loa mapping.

Reply:
I believe that this should not ignore client's acr-to-loa mapping TBH. IMO the `default_acr_values` available of client should allow same values like allowed in the `acr_values` (or `claims`) parameter sent to the client.

For example in case that client acr-to-loa mapping contains mappings like `silver -> 1`, then it is valid for client to send `acr_values` parameter with value `silver` in OIDC request. Similarly I consider it appropriate for the client to use `silver` as one of the values in the `default acr values` .

Reply:
The fact that `acr_values_supported` from well-known endpoint does not consider client's acr-to-loa mapping is due the fact that well-known endpoint is scoped to the realm. So I don't see a way to consider any client's acr-to-loa mapping in this endpoint as there is not any client available.

But in this validation, we have client available and hence client mapping can be used.

We discussed earlier within the team that we should probably never introduce option for `ACR-To-Loa` mapping configured at client level as OIDC also doesn't have this and realm mapping is usually completely sufficient. However the option was introduced back in the initial step-up PR and hence we need to consider it though...

Reply:
@mposolda Thank you for your clarification. I agree on your current specification.